zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java

```diff
@@ -89,10 +89,9 @@ public HAccountResetPasswordKey requestPasswordReset(
             + account.getPerson().getName() + System.currentTimeMillis()));
 
         account.setAccountResetPasswordKey(key);
-        accountResetPasswordKeyDAO.makePersistent(key);
+        key = accountResetPasswordKeyDAO.makePersistent(key);
+        accountResetPasswordKeyDAO.flush();
```

Comment:
I think the problem may be lacking a ```@Transactional``` on the method. After CDI migration, we need to explicitly add transactional on each publicly invoked  method (e.g. invoked from jsf). Seam opens a transaction implicitly for each request so we get away with not having the annotation.